code/modules/admin/verbs/modifyvariables.dm

```diff
@@ -130,16 +130,40 @@ var/list/forbidden_varedit_object_types = list(
 
 	if(!istype(L,/list)) src << "Not a List."
 
-	var/list/locked = list("vars", "key", "ckey", "client", "firemut", "ishulk", "telekinesis", "xray", "virus", "viruses", "cuffed", "ka", "last_eaten", "urine", "poo", "icon", "icon_state")
-	var/list/names = sortList(L)
+	if(L.len > 1000)
+		var/confirm = alert(src, "The list you're trying to edit is very long, continuing may crash the server.", "Warning", "Continue", "Abort")
+		if(confirm != "Continue")
+			return
+
+	var/list/locked = list("vars", "key", "ckey", "client", "firemut", "ishulk", "telekinesis", "xray", "virus", "viruses", "cuffed", "ka", "last_eaten", "urine", "poo", "icon", "icon_state", "step_x", "step_y")

```

Comment:
Some of these are blacklisted for editing so that anyone who may have R_VAREDIT but not R_DEBUG won't varedit something they're not supposed to. key and ckey are there because messing with those can cause client crashes, a lot of these vars are probably legacy code though.